### PR TITLE
[Merged by Bors] - Document sub apps

### PR DESCRIPTION
**Pull request by Nilirad**
Documentation added to:
- `App::add_sub_app` (
- `App::update` (mentions that sub apps are updated here)

### Future work
- An example for `add_sub_app` would be good, but I wasn't able to come up with a simple one.
- Since `SubApp` is private, maybe the concept of sub applications could be introduced in the `App` struct-level documentation.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
> Since SubApp is private, maybe the concept of sub applications could be introduced in the App struct-level documentation.

I would prefer to hold off until we have a user-ready multiple worlds API, for now it's just an implementation detail.

File: crates/bevy_app/src/app.rs
Comment (alice-i-cecile) [inline crates/bevy_app/src/app.rs:89]: IMO this added line should go in the body, rather than the tooltip.

It's not essential information, but is a useful detail.

File: crates/bevy_app/src/app.rs
Comment (alice-i-cecile) [inline crates/bevy_app/src/app.rs:842]: > The `World` of the sub app is the same of the main app.
 
 Is this true? I thought we had to copy data for entities between the main world and render world?

Comment (alice-i-cecile) [inline crates/bevy_app/src/app.rs:842]:
```suggestion
    /// The `World` of the sub app is the same of the main app. The provided function `f` runs each
```

Comment (Nilirad) [reply crates/bevy_app/src/app.rs:842]: Yeah you're right, I got confused. The subapp has its own world, however, the main world is passed as an argument to the World parameter of `f`, so the two can interact.

**Comment by Nilirad** [comment]
I applied the suggestions, added info about the parameters of `f`, reworded a little bit, and I linked `add_sub_app` from `update` for more info about sub apps, especially since the meaning of sub app's “runner” function's parameters is determined by `update`. I hope this may help to keep docs updated in case of a change.

I think that the best thing to solve this problem is to make a public “trait alias“ for `Fn(&mut World, &mut App)`. This way the documentation about function parameters is coupled to the trait itself. Unfortunately, trait aliases are still unstable. There are ways to hack around it e.g. by making a new trait + blanket impl, but I don't know if it is worth the effort.

**Comment by alice-i-cecile** [comment]
bors try

**Comment by alice-i-cecile** [comment]
@Nilirad could you rebase this? I merged in some very simple docs for this recently; I suspect that yours will mostly supplant them.

**Comment by Nilirad** [comment]
@alice-i-cecile Done. I completely reverted back the `add_sub_app` docs to my version, since the description and the details seem more objective.

I have mixed feelings about saying that sub apps are “experimental” though. It is certainly true, but I don't know how worth is to add this: for example, there are lots of other features that are subject to change but aren't marked as experimental.

I also removed the description about `SubApp`: revealing its existence and its fields would defeat the reasons that made it a private API.

**Comment by alice-i-cecile** [comment]
> I have mixed feelings about saying that sub apps are “experimental” though. It is certainly true, but I don't know how worth it is to add this: for example, there are lots of other features that are subject to change but aren't marked as experimental.

SubApps are... unusually half-baked. They really needed an RFC process, and a thoughtful public-facing API, but getting *something* in place for the rendering was essential.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_app/src/app.rs
Comment (NiklasEi) [inline crates/bevy_app/src/app.rs:859]: Can we rename `f` to something descriptive? E.g. `sub_app_runner`

**Review by NiklasEi — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@Nilirad once that rename is complete, I'm happy to merge this in :)

**Comment by Nilirad** [comment]
@alice-i-cecile Done!

**Review by alice-i-cecile — APPROVED** [review]

**Pull request closed without merging**
